src/libtoast/math/toast_qarray.cpp

```diff
@@ -794,3 +794,47 @@ void toast::qarray::to_angles ( size_t n, double const * quat, double * theta,
 }
 
 
+// Convert quaternions to latitude and longitude
+
+void toast::qarray::to_position ( size_t n, double const * quat, double * theta,
+    double * phi ) {
+
+    double const zaxis[3] = { 0.0, 0.0, 1.0 };
+
+    #pragma omp parallel default(shared)
+    {
+        size_t qf;
+
+        double norm;
+
+        double dir[3];
+        double qtemp[4];
+
+        #pragma omp for schedule(static)
+        for ( size_t i = 0; i < n; ++i ) {
+            qf = 4 * i;
+
+            norm = 0.0;
+            norm += quat[qf] * quat[qf];
+            norm += quat[qf+1] * quat[qf+1];
+            norm += quat[qf+2] * quat[qf+2];
+            norm += quat[qf+3] * quat[qf+3];
+            norm = 1.0 / ::sqrt ( norm );
+            qtemp[0] = quat[qf] * norm;
+            qtemp[1] = quat[qf+1] * norm;
+            qtemp[2] = quat[qf+2] * norm;
+            qtemp[3] = quat[qf+3] * norm;
+
+            toast::qarray::rotate ( 1, qtemp, 1, zaxis, dir );
+
+            theta[i] = toast::PI_2 - ::asin ( dir[2] );
+            phi[i] = ::atan2 ( dir[1], dir[0] );
```

Comment:
what about doing `atan2( -dir[1], -dir[0] ) + pi` to avoid branching? https://stackoverflow.com/questions/1311049/how-to-map-atan2-to-degrees-0-360

Reply:
@zonca if we are optimizing this method, I'd take a hard look at the need to normalize the supplied quaternions. For now, it is enough to repeat what is done in to_angles.